frame/staking/src/mock.rs

```diff
@@ -443,8 +447,6 @@ pub fn advance_session() {
 }
 
 pub fn start_session(session_index: SessionIndex) {
-	// Compensate for session delay
-	let session_index = session_index + 1;
```

Comment:
this is one reason why tests with start_session has been modified, it makes them clearer IMHO, also most tests don't use this but start_era instead.